binstall.js

```diff
@@ -48,29 +49,16 @@ function untgz(url, path, options) {
       if (error.code !== 'EEXIST') throw error;
     }
 
-    request
-      .get(url, function(error, response) {
-        if (error) {
-          reject("Error communicating with URL " + url + " " + error);
-          return;
-        }
-        if (response.statusCode == 404) {
-          var errorMessage = options.errorMessage || "Not Found: " + url;
-
-          reject(new Error(errorMessage));
-          return;
-        }
-
-        if (verbose) {
-          console.log("Downloading binaries from " + url);
-        }
-
-        response.on("error", function() {
-          reject("Error receiving " + url);
-        });
-      })
+    if (verbose) {
+      console.log("Downloading binaries from " + url);
+    }
+    return fetch(url)
+    .then(function (response) {
+      response.body
       .pipe(gunzip)
       .pipe(untar);
+      return
```

Comment:
Don't we need to return the stream or the promise or whatever so that the caller will be able to wait for the operation to fully complete?  (Or is that happening somehow as this is?)